ambry-quota/src/main/java/com/github/ambry/quota/capacityunit/CapacityUnit.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2022 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.quota.capacityunit;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.ambry.quota.QuotaName;
+import com.google.common.util.concurrent.AtomicDouble;
+import org.codehaus.jackson.annotate.JsonIgnore;
+
+
+/**
+ * Class encapsulating the Read_Capacity_Unit and Write_Capacity_Unit quotas or usage for resource.
+ */
+public class CapacityUnit {
+  static final String RCU_FIELD_NAME = "rcu";
+  static final String WCU_FIELD_NAME = "wcu";
+
+  private final AtomicDouble rcu;
+  private final AtomicDouble wcu;
```

Comment:
why do we need an AtomicDouble for the RCU and WCU? Why are immutable once they are loaded from configuration file. We can just use a long type here.

Reply:
The instances of CapacityUnit class can represent the quota value or the usage.
There will be a refresh flow in the (future) CUQuotaSource that will periodically refresh the usage of every QuotaResource to 0, to make sure that the usage isn't always monotonically increasing. Since this refresh can happen while requests attempt to read the usage values, we need atomic for thread safety.

Reply:
then let's make it AtomicLong instead of AtomicDouble.

Reply:
done